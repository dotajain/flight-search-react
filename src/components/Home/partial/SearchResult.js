import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { connect } from 'react-redux';

import SearchBreadcrumb from './SearchBreadcrumb';
import SearchResultCard from './SearchResultCard';

const SearchResult = ({ searchResult, range }) => {
  if (_.isEmpty(searchResult)) {
    return <div className="col col--9 no-result" />;
  }

  const resultItems = {
    returnFlight: null,
    onWardFlight: null,
  };

  _.map(searchResult, result => {
    if (result.result.type === 'return') {
      const rangeData = [];

      _.map(result.result.data, item => {
        if (item.fare > range.min && item.fare < range.max) {
          rangeData.push(item);
        }
      });

      const orderedByFare = _.orderBy(rangeData, ['fare'], ['asc']);
      resultItems.returnFlight = orderedByFare;
    }
    if (result.result.type === 'onward') {
      const rangeData = [];

      _.map(result.result.data, item => {
        if (item.fare > range.min && item.fare < range.max) {
          rangeData.push(item);
        }
      });

      const orderedByFare = _.orderBy(rangeData, ['fare'], ['asc']);
      resultItems.onWardFlight = orderedByFare;
    }
  });

  const filteredResult = [];
  _.map(resultItems.onWardFlight, (item, i) => {
    const data = {
      returnFlight: _.isEmpty(resultItems.returnFlight)
        ? null
        : resultItems.returnFlight[i],
      onWardFlight: item,
    };
    filteredResult.push(data);
  });

  const searchCard = _.map(filteredResult, (item, i) => (
    <SearchResultCard key={i} result={item} />
  ));

  return (
    <div className="col col--9">
      <div className="box">
        <SearchBreadcrumb />

        <div className="search__result">{searchCard}</div>
      </div>
    </div>
  );
};

SearchResult.propTypes = {
  searchResult: PropTypes.array,
  range: PropTypes.object,
};

const mapStateToProps = state => ({
  searchResult: state.flightSearch.searchResult,
  range: state.flightSearch.range,
});
export default connect(mapStateToProps)(SearchResult);
