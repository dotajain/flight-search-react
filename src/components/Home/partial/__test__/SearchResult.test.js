import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../store';

import SearchResult from '../SearchResult';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered SearchResult Component', () => {
  const component = renderer.create(<Provider store={store}><SearchResult /></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
