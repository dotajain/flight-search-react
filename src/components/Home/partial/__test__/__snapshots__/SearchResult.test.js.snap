// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered SearchResult Component 1`] = `
<div
  className="col col--9 no-result"
/>
`;
